### PR TITLE
MLIBZ-2661: bugfix for large pull memory consumption

### PR DESCRIPTION
**Pull request by heyzooi**
#### Description

Realm have a limitation of keeping all objects in memory during a single large transaction. To avoid this situation we did a few changes around the cache layer.

#### Changes

- Request Start persisted is now from the `count()` 

#### Tests

- A few unit tests to ensure the cache data still consistent

### CODE REVIEW

**Comment by ivanstoyanov** [comment]
Would this limitation also apply to Android? Can we make the same test there?

File: Kinvey/KinveyTests/NetworkStoreTests.swift [hunk 1]
```diff
@@ -771,6 +771,48 @@ class NetworkStoreTests: StoreTestCase {
         }
     }
     
+    func testTranslateQueryCodable() {
+        let store = try! DataStore<PersonCodable>.collection(.sync)
```
Comment (vinaygahlawat) [inline Kinvey/KinveyTests/NetworkStoreTests.swift]: Why is a `(.sync)` store type in the `NetworkStoreTests` file?

Comment (heyzooi) [reply Kinvey/KinveyTests/NetworkStoreTests.swift]: good call, let me move that do the `SyncStoreTests`

**Comment by heyzooi** [comment]
@ivanstoyanov for sure worth to test, but I would guess it also happen in Android since the code base is the same.

**Pull request merged**